sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ContainerAPITest.groovy

```diff
@@ -1984,6 +1985,99 @@ class ContainerAPITest extends APISpec {
         response.getHeaders().getValue("x-ms-version") == "2017-11-09"
     }
 
+    def "create if not exists"() {
+        setup:
+        def cc = primaryBlobServiceClient.getBlobContainerClient(generateContainerName())
+
+        when:
+        cc.createIfNotExists()
+
+        then:
+        cc.exists()
+    }
+
+    def "create if not exists with response"() {
+        setup:
+        def cc = primaryBlobServiceClient.getBlobContainerClient(generateContainerName())
+
+        when:
+        def response = cc.createIfNotExistsWithResponse(null, null, null, null)
+
+        then:
+        cc.exists()
+        response != null
+        response.getStatusCode() == 201
+    }
+
+    def "Create if not exists on a container that already exists"() {
+        setup:
+        def cc = primaryBlobServiceClient.getBlobContainerClient(generateContainerName())
+        def initialResponse = cc.createIfNotExistsWithResponse(null, null, null, null)
+
+        when:
+        def secondResponse = cc.createIfNotExistsWithResponse(null, null, null, null)
+
+        then:
+        initialResponse.getStatusCode() == 201
+        secondResponse == null
+    }
+
+    def "Delete container that exists"() {
+        setup:
+        def cc = primaryBlobServiceClient.getBlobContainerClient(generateContainerName())
+        cc.create()
+
+        when:
+        cc.deleteIfExists()
+
+        then:
+        cc.exists() == false
+    }
+
+    def "Delete container that exists with response"() {
+        setup:
+        def cc = primaryBlobServiceClient.getBlobContainerClient(generateContainerName())
+        cc.create()
+
+        when:
+        def response = cc.deleteIfExistsWithResponse(new BlobRequestConditions(), null, null)
+
+        then:
+        response != null
+        response.getStatusCode() == 202
+        cc.exists() == false
+    }
+
+    def "Delete container that was already deleted"() {
```

Comment:
This test shouldn't be necessary because it should be the same response from the service whether the container was previously deleted or just never created